source/troubleshooting.txt

```diff
@@ -122,4 +122,305 @@ replica set members. However, you can't access these addresses and ports through
+   `PYTHON-3406 <https://jira.mongodb.org/browse/PYTHON-3406>`__ in Jira.
+   
+   For more information about the problems caused by Python locks in
+   multithreaded contexts with ``fork()``, see `Issue 6721 <http://bugs.python.org/issue6721>`__
+   in the Python Issue Tracker.
+
+Why Does My Query Work in the Shell But Not in {+driver-short+}?
+-------------------------------------------------------
+
+After the ``_id`` field, which is always first, the key-value pairs in a BSON document can
+be in any order. The ``mongo`` shell preserves key order when reading and writing
+data, as shown by the fields "b" and "a" in the following code example:
+
+.. code-block:: javascript
+
+   > // mongo shell.
+   > db.collection.insertOne( { "_id" : 1, "subdocument" : { "b" : 1, "a" : 1 } } )
+   WriteResult({ "nInserted" : 1 })
+   > db.collection.findOne()
+   { "_id" : 1, "subdocument" : { "b" : 1, "a" : 1 } }
+
+{+driver-short+} represents BSON documents as Python dictionaries by default,
+and the order of keys in dictionaries is not defined. In Python, a dictionary declared with
+the "a" key first is the same as one with the "b" key first. In the following example,
+the keys are displayed in the same order regardless of their order in the ``print``
+statement:
+
+.. code-block:: python
+   
+   >>> print({'a': 1.0, 'b': 1.0})
+   {'a': 1.0, 'b': 1.0}
+   >>> print({'b': 1.0, 'a': 1.0})
+   {'a': 1.0, 'b': 1.0}
+
+Similarly, Python dictionaries might not show keys in the order they are
+stored in BSON. The following example shows the result of printing the document
+inserted in a preceding example:
+
+.. code-block:: python
+
+   >>> print(collection.find_one())
+   {'_id': 1.0, 'subdocument': {'a': 1.0, 'b': 1.0}}
+
+To preserve the order of keys when reading BSON, use the ``SON`` class,
+which is a dictionary that remembers its key order.
+
+The following code example shows how to create a collection
+configured to use the ``SON`` class:
+
+.. code-block:: python
+
+   >>> from bson import CodecOptions, SON
+   >>> opts = CodecOptions(document_class=SON)
+   >>> opts
+   CodecOptions(document_class=...SON..., tz_aware=False, uuid_representation=UuidRepresentation.UNSPECIFIED, unicode_decode_error_handler='strict', tzinfo=None, type_registry=TypeRegistry(type_codecs=[], fallback_encoder=None), datetime_conversion=DatetimeConversion.DATETIME)
+   >>> collection_son = collection.with_options(codec_options=opts)
+
+When you find the preceding subdocument, query results are represented with
+``SON`` objects and key order is preserved:
+
+.. code-block:: python
+
+   >>> print(collection_son.find_one())
+   SON([('_id', 1.0), ('subdocument', SON([('b', 1.0), ('a', 1.0)]))])
+
+The subdocument's actual storage layout is now visible: "b" is before "a".
+
+Because a Python dictionary's key order is not defined, you cannot predict how it will be
+serialized *to* BSON. But MongoDB considers subdocuments equal only if their
```

Comment:
```suggestion
Because a Python dictionary's key order is not defined, you cannot predict how it will be
serialized to BSON, but MongoDB considers subdocuments equal only if their
```